app/controllers/users_controller.rb

```diff
@@ -18,11 +18,24 @@ def new_with_invitation_token
 
   def create
     @user = User.new(user_params)
-    if @user.save
-      handle_invitation
-      AppMailer.delay.welcome(@user)
-      redirect_to sign_in_path
+    if @user.valid?
+      charge = StripeWrapper::Charge.create(
+        amount: 999,
+        source: params[:stripeToken],
+        description: "Sign up charge for #{@user.email}"
+      )
+      if charge.successful?
+        @user.save
+        handle_invitation
+        flash[:success] = "Account Created!"
+        AppMailer.delay.welcome(@user)
+        redirect_to sign_in_path
+      else
+        flash[:danger] = charge.error_message
```

Comment:
When you render, I would recommend using `flash.now` otherwise it pops the message on next request as well.